server/controllers/pages.js

```diff
@@ -16,7 +16,27 @@ const entryHelper = require('../helpers/entry')
  * Edit document in Markdown
  */
 router.get('/edit/*', (req, res, next) => {
-  if (!res.locals.rights.write) {
```

Comment:
Keep this check as well initially. If this check passes, then pnly we try to compute xpath logic. So:

```
if(!res.local.rights.write) return error_forbidden;
...// your code here
if(!admin && ... // the final if clause as it is now)
```